engine/common/requester/engine.go

```diff
@@ -362,12 +366,25 @@ func (e *Engine) dispatchRequest() (bool, error) {
 		Nonce:     rand.Uint64(),
 		EntityIDs: entityIDs,
 	}
+
+	requestStart := time.Now()
+	e.log.Debug().
+		Hex("provider", logging.ID(providerID)).
+		Int("num_selected", len(entityIDs)).
+		Msg("sending entity request")
+
 	err = e.con.Unicast(req, providerID)
 	if err != nil {
 		return true, fmt.Errorf("could not send request: %w", err)
 	}
 	e.requests[req.Nonce] = req
 
+	e.log.Debug().
+		Hex("provider", logging.ID(providerID)).
+		Int("num_selected", len(entityIDs)).
+		TimeDiff("duration", time.Now(), requestStart).
+		Msg("entity request sent")
+
 	// NOTE: we forget about requests after the expiry of the shortest retry time
 	// from the entities in the list; this means that we purge requests aggressively.
 	// However, most requests should be responded to on the first attempt and clearing
```

Comment:
Added entity IDs to both requests and responses (along with the nonce for the ease of grepping).  While here, since it is more heavyweight now, hidden it behind the `e.log.Debug().Enabled()`.